### ISSUE TITLE
Add Survivor's Rollerblades

### ISSUE THREAD

**Issue by Hague** [CleverRaven/Cataclysm-DDA#16679]
Survivor's rollerblades are pairs of armored, environmentally-resistant rollerblades designed much in the same vein as other survivor's gear. The special caveat however, is that the in-line skates have been inset into a platform sole that allows a spring-loaded mechanism to retract them in and out, allowing a canny survivor to quickly switch between walking and skating. They are basically a slightly heavier set of light survivor boots with an activation function that swaps the skating qualities from the boots, allowing a player to swap between skating and walking with a minor delay.

Requirements:
Skill used - Fabrication Diff - 5 
Requirements (Tailoring 5, base of light survivor boots, Mechanics 3)
Cutting Quality 1
Metal Sawing 1
Screwdriving Quality 1 (to disassemble rollerblades)
Welder, integrated toolset, or soldering iron (12 charges)
Materials:
1 pair of rollerblades
1 pair of light survivor boots
2 kevlar plates
4 scrap metal
2 springs
2 super glue
Concept is you break down the rollerblades, cut a sole-liner from the 2 kevlar plates, super glue them to the original sole of the survivor boots, inset the rail line from the blades into a carved channel, a spring and tension rod is installed to rotate and pop out the skates and a lever arm locks the rail in place (hence mechanics requirement)

**Comment by Coolthulhu** [comment]
Some silliness here and there is one thing, but survivor rollerblades?

**Comment by mugling** [comment]
I'm not sure about the general trend towards an improved *survivor* version of everything even where there aren't any meaningful modifications that could be expected to be made

**Comment by Hague** [comment]
Rollerblades are unarmored but they are useful tools. The biggest problem with them is alternating between flat and rough terrain (your move takes a big hit) and removing them takes forever, this is a problem-solver's solution to a potentially deadly problem with them. They are armored rollerblades that provide a faster means of switching between walking and skating. It's not really that silly. A similar concept already exist as [commercial sneakers.](http://www.heelys.com/images/slideshow/image3.jpg)

Not any more ridiculous than some of the things we have in this game: Blob car parts, tiny plutonium powered knife robots that have batteries, plutonium cells, and solar panels in a tiny chassis that floats forever, or japanese wooden training swords that hit twice as hard as a baseball bat, magical artifacts that summon monsters that melt away into nothing... Yup, armored rollerblades soooo crazy.

**Comment by mugling** [comment]
> Blob car parts

Mod content

> plutonium cells

Future science

> japanese wooden training swords that hit twice as hard as a baseball bat

Bug? Consider PR

**Comment by Labtop-215** [comment]
> Blob car parts, tiny plutonium powered knife robots that have batteries, plutonium cells, and solar panels in a tiny chassis that floats forever, or japanese wooden training swords that hit twice as hard as a baseball bat, magical artifacts that summon monsters that melt away into nothing... Yup, armored rollerblades soooo crazy.

Unless Blob car parts have been added as a mod, I've never seen them.  The manhacks are the day after tomorrows equivalent of drones, so no they arn't crazy.  The Bokken dosn't hit twice as hard as a baseball bat, and it's also pretty rare.  But if it did hit twice as hard reguardless of skill, then you could make a case that it is unbalanced.

Artifacts are also kind of a placeholder from the days that Whales was developing Cataclysm.

However, I would like to point out that this would break the game pretty badly, because it would let you travel super fast on solid pavement, and then run like normal on dirt and grass afterwards.  Basically turning roadways into the equivalent of a river in Sid Meier's Alpha Centauri.

At least with roller skates or blades, you are road-bound if you choose to use that to outrun zombies.

**Comment by Coolthulhu** [comment]
> tiny plutonium powered knife robots that have batteries, plutonium cells, and solar panels in a tiny chassis that floats forever

Simplification. Having them go dormant would be better.

> japanese wooden training swords that hit twice as hard as a baseball bat

This one is quite stupid, feel free to PR a nerf

> magical artifacts that summon monsters that melt away into nothing

Suspension of disbelief doesn't mean "no things that aren't real", it means "no things that are real but work unlike the real thing".

> Yup, armored rollerblades soooo crazy.

Yes, yes they are. They're somewhere around the sanity level of:
* Survivor bread (no, not wastebread)
* Survivor comic books
* Survivor coffee maker
* Survivor vibrator

Then finally, the whole simplification with rollerblades giving a 2x speed boost on roads is also a simplification and I wouldn't merge any PR that adds more rollerblades before that was turned into something sane (ie. saving on stamina instead of speed boost).

**Comment by Hague** [comment]
That's not a fair characterization at all. Armoring all of those items doesn't provide a benefit. Armoring skates is a valid use. You freakin' wear them and can engage in combat wearing them. You're being deliberately obtuse.

As for balance, just make them pretty damn encumbering, they are basically platform boots, running in them would be tough, but not as tough as running in skates or rolling over dirt.

Alternate version would be an inline- skate rail you can clip into an accompanying rail in modified pair of boots. Clip-on skates that take less time to swap back and forth.

As for the speed thing, yes, you do go significantly faster while using rollerblades in real life. Skating needs a lot of balance, so it should require more balance checks than it does now: making ranged attacks (spear kiting would be harder) and when sprinting would be ideal choices for extra balance checks.

**Comment by Labtop-215** [comment]
> That's not a fair characterization at all. Armoring all of those items doesn't provide a benefit.
"Survivor" objects are objects that the player has either created or modified in order to provide some benefit over using a regular item.  You wouldn't say that a survivor mess kit is just an armored mess kit, for instance.  What Coolthulu is trying to get at here is that you are trying to create a "Survivor" luxury item.

Also

> As for balance, just make them pretty damn encumbering, they are basically platform boots, running in them would be tough, but not as tough as running in skates or rolling over dirt.

Encumbering wouldn't even begin to describe it, they would either be unreliable, as the metal obscuring the roller-skates would potentially slip and cause you to slip when you would least expect it, or they would be like lead weighted blocks on your feet.  If they remain "armored" after the conversion, then they would become to heavy and en-cumbersome to use for either skating or walking.

For clip on skates, I'm guess you would apply the rail boots to turn the skates into non skates?  At that point wouldn't it just be better anyway, to take off your skates and put on your regular boots?  The only benefit of these skates at that point would be a temporary volume increase when those to objects are combined.

**Comment by Hague** [comment]
They're only armored as much as light survivor boots already are. The extra kevlar is just an expense of expanding the thickness of the sole. 

Rollerblades have base encumberment of 25. Combat boots have the same. Hiking boots have 12 encumberment, a rough estimate of the "boot" part of a set of in-line skates. Split the difference: 13 encumberment to add skates to any set of footwear. Tack 13 encumberment onto Light survivor boot's base encumberment of 18 to get 31. Don't improve the armor at all, keep it the same. 

Also, the clip-on skate rails could be attached rather easily. Slides into a rail recessed into a hardened sole accessed via a gap in the back the heel. Just lift your leg backwards and slip the skate into the rail and it clips into place. To remove, reach down, pull a spring-tension pin and kick against a hard surface to have them slide back out. Definitely a lot faster than putting on a whole pair of boots and lacing them up. (I dare you to skate without lacing up!) That's similar how "Heely's" work, you slip a spring tension axle- and wheel into a compartment in the heel. There's people who do conversion mods to have swappable rollerblades and ice skates for their boots. The reason I didn't suggest this was because it seemed like it would take more work to code than just having boots that would swap back and forth.

**Comment by baraba3** [comment]
i wouldnt mind having nomex/ceramic insulated  survivor skates in game for better protection from acid as currently you cant improve acid protection with tailor kit.Armoring of other kinds can be already done with  tailor kit.
.
having skate with ability to swithc wheels on /off  on surfaces seems too much of bonus.skates should force you on flat surfaces.

**Comment by i2amroy** [comment]
Rather than just going the "better" route for retractable rollerblades we could always go for the "more flexible but not as good as either in it's specific thing" route instead. I mean it's not really believable that those little rollerskate shoes you see kids wearing are gonna hit the same speeds that normal roller blades do (or that you are somehow going to be able to cobble together rollerblades that follow that paradigm and can), but they will be able to switch back and forth at will, letting you have a smaller bonus than straight rollerblades without needing to suffer penalties offroading like they would normally hit you with. Game wise it gives you a bit of a choice too, do you take the pure rollerblades and go for straight speed (but risk getting screwed if you need to run through a field) or do you take a more flexible one that gives you the option to go through that field if you need to but doesn't provide as large of a boost on the flat ground?

>Then finally, the whole simplification with rollerblades giving a 2x speed boost on roads is also a simplification and I wouldn't merge any PR that adds more rollerblades before that was turned into something sane (ie. saving on stamina instead of speed boost).

Looking at some actual numbers for jogging/running/rollerblading a much more realistic value would be a 1.4x speed boost and 70% stamina consumption. The numbers get screwed up slightly because our "walking" speed is only 1/2 our "running" speed, while in real life "walking" tends to range around 1/3-1/4th of "running" speed, but 1.4x speed and 70% stamina is approximately close to what the numbers in the real world for semi-skilled people look like. (Of course competitive racing road-course rollerbladers *can* actually hit around 2x the speed of competitive marathon runners, but for the average joe 1.4x and a stamina reduction is going to be much more realistic.)

**Comment by harald921** [comment]
Just skimmed through, but here's my input: Armored rollerblades seems really overpowered. But how about adding either a craftable version of the rollerblades that is fairly encumbering, not very well armored but that you can quickly remove the skates?

Or how about rollerskates? The ones with four wheels? Basically slower rollerblades you can quickly take off and probably wear over your shoes.

**Comment by Coolthulhu** [comment]
> Armored rollerblades seems really overpowered.

It's not the armor that's overpowered, it's the core mechanics of rollerblades that don't make sense and honestly should be treated as a strict blocker of any rollerblade-related additions that aren't just nerfs.

The primary use of rollerblades doesn't make any sense: attacking, running away, quickly turning back for a quick reach attack/shot, running away and so on would not work well with rollerblades.

> Looking at some actual numbers for jogging/running/rollerblading a much more realistic value would be a 1.4x speed boost and 70% stamina consumption.

That's in one direction. Without that assumption, there should be no speed boost.

**Comment by i2amroy** [comment]
>That's in one direction. Without that assumption, there should be no speed boost.

You could also flip the assumption and have there be *only* the 2x speed boost and no stamina reduction, which is also still totally realistic ("very slow" rollerblading speed for people actually going places tends to be about 6 mph, approximately 2x average walking speed, while the running example, as I mentioned earlier, has 2x speed completely obtainable [though you're burning approximately the same amount of calories at that point and working just as hard]).

I figured that taking a middle path of the two would still be realistic while being much easier to bring in-line with balance as opposed to the alternatives of just keeping the 2x speed factor with no stamina reduction or by making running essentially free (which is basically the same as a 2x speed factor except it doesn't let you push up to 4x speed at the cost of stamina like a basic 2x speed factor does).

I mean if we really wanted to be realistic then the ideal rollerblades would have three "movement types" instead of just running and walking; there would be walking which would get a fairly large speed boost (and still not consume stamina), "jogging" which would only get a small speed boost and would have vastly reduced stamina costs, and "running" which would get a large speed boost again but no stamina cost reductions.

---
Alternative Idea:
Make rollerblades require you to move in a fairly straight line to maintain their speed boost with a "get going" time of 2-3 tiles. The first step taken in a direction is a 1x speed, then the next at 1.5x, and then the last one at 2x speed. Taking a step at a diagonal in your direction of travel maintains the current bonus without reducing it (thus letting you take curved corners while maintaining your speed), taking a perpendicular step reduces your speed by one level for that step (so 2x->1.5x->1x) and taking any steps backwards/diagonally backwards from your direction of travel or waiting in place would drop you back down to the 1x speed. (Ideally attack directions should also influence your speed the same as moving, but it probably would be fine without that). That would model the realistic speed attributes of rollerblading in terms of traveling in fairly straight lines, and would also (realistically) stop the back-and-forth hit-dancing that they are often abused for.

**Comment by Hague** [comment]
Combat on rollerblades isn't modeled in a very complicated way. You make a dodge roll to avoid falling on your ass when you get attacked. Without high dodge, you're a liability in combat. Skater profession gives you a trait that lets you make that roll easier (not guaranteed) Certain weapons would logically be difficult to use: high-recoil weapons (though, I think it might be pretty funny to be pushed in the opposite direction you fire. Rocket pack, anyone?) and reach weapons would be the primary offenders. One way to model a better balance check for melee combat on skates is to throw a balance check (dodge roll + appropriate bonuses) melee attacks that are completely blocked by armor or on stuck-ins. This would represent the player's momentum in attack being halted by a solid target. While this wouldn't completely prevent hit and run attacks, it would provide opportunity for failure when you encountering appropriately armored enemies or when poorly trained in your weapons. Also, a boomer exploding nearby when on unstable footing (while on skates or standing on rubble/debris) should also give a balance check.

Hit and run tactics would absolutely be the way you use skates in a combat scenario, move in at speed, strike and get away, repeat. It'd be like saying a Kreck doesn't fight fair because it doesn't stand there and tank your hits. The same behavior is modeled by manhacks. Also, trained skaters can skate backward all day long, so jabbing at someone with a spear is just an exercise in balance that just needs to be modeled better through rolls. A superhuman with 14/14/14/14 stats and 12 dodge would be able to skate backwards and jab stuff with a spear just fine. Probably sing a song while doing it. Factor in stamina as a debuff to your balance and you could get overwhelmed on skates rather easily if you are unskilled.

**Comment by i2amroy** [comment]
>Hit and run tactics would absolutely be the way you use skates in a combat scenario, move in at speed, strike and get away, repeat.

Yeah, but with real skates you'd do it by zooming *past* something and striking it as you passed by (which has the limiting factor that there can't just be a crowd of other enemies on the other side of the target) and then looping back for another pass, not the constantly reversing direction method that the game currently encourages that lets you hit and then safely retreat in the direction of your choosing without any real limitations at all (with a fast enough weapon the current balancing lets you treat just about any patch of flat ground the same way that you would a bush in terms of hit and run combat).

**Comment by Hague** [comment]
Combat doesn't specify. How do we know what facing your character has? Are they backing up slowly while fighting. Obviously you wouldn't turn your back on zombies on foot. Again it's all about balance

Combat rounds are what? 6 seconds long? Someone can execute a turn on skates in 6 seconds easily. Direction of motion isn't modeled in the game or we'd have to keep running a few extra steps just top stop ourselves after sprinting normally. The turns are fudged in with the motion. Like I said, add more balance checks and chances to screw up and it will be fine. I mean, you have to make a roll every few turns while driving? Why not a skill roll while skating? Seems fair enough to me. Screw up and you move the wrong direction instead of the one you intended.

**Comment by harald921** [comment]
Honestly, I believe that while on skates you should be practically unable to fight anything significant. I mean, really it should have a _severe_ impact on both melee and dodging

**Comment by Coolthulhu** [comment]
> Honestly, I believe that while on skates you should be practically unable to fight anything significant. I mean, really it should have a severe impact on both melee and dodging

Pretty much this.
At best using guns with moderate precision - certainly not as accurately as with stable footing.

> Screw up and you move the wrong direction instead of the one you intended.

This sounds like a bad idea. We already have direction changing for driving and it's annoying and most of the time only happens when it doesn't matter.

**Comment by Hague** [comment]
Clearly you guys don't watch much Roller Derby or Hockey? Looks like those guys do pretty well for the most part. I can only imagine what they could do if they were _supposed_ to hurt the opposition.

The directional movement as part of a screw-up was just reaching towards what already exists for a foundation of a rule-base. That said, balance is my primary go-to for reworking skating. More rigorous balance checks for all kinds of combat on skates. Remember how nerfed dodge gets while encumbered. If you're keeping unencumbered to fight on skates, you're taking more of a risk when you eventually flub a roll and lose your balance (for one reason or another). Too encumbered and you should regularly lose your balance.

I'm not even sure you can have negative dodge value, but when it comes to any kind of balancing performance by the player, there should be something that makes excessive encumbrance more of an issue if it isn't already modeled.
